Player/Player.gd

```diff
@@ -19,6 +19,7 @@ func _ready() -> void:
 	$HUD.player = self
 	for weapon in weapon_holder.get_children():
 		weapon.connect("recoil", self, "_on_weapon_recoil")
+		weapon.connect("sniper_raise",self,"_on_sniper_raise")
```

Comment:
this should be two generic signals "camera_zoom(amount)" and "camera_unzoom" that can be potentially used by other weapons or systems.